.github/workflows/release.yml

```diff
@@ -0,0 +1,74 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+
+env:
+  GOPROXY: https://proxy.golang.org/
+  PROTOCOL_VERSION: "6.0"
+  PROTOCOL_VERSIONS: "6.0"
+  SIGNER: interim_signing_subkey_7685B676
+
+jobs:
+  Release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Read go version
+        id: go-version
+        run: |
+          content=`cat ./.go-version`
+          echo "::set-output name=content::$content"
+      - uses: actions/setup-go@v2
+        with:
+          # TODO: Replace with go-version-from-file when it is supported
+          # https://github.com/actions/setup-go/pull/62
+          go-version: ${{ steps.go-version.outputs.content }}
+      - uses: hashicorp/setup-hc-releases@v1
+        with:
+          github-token: ${{ secrets.HC_RELEASES_TOKEN }}
+          signer: ${{ env.SIGNER }}
+      - uses: hashicorp/setup-signore@v2
+        with:
+          github-token: ${{ secrets.SETUP_SIGNORE_GITHUB_TOKEN }}
+          signer: interim_signing_subkey_768B676
+      - name: Release Notes
+        run: sed -n -e "1{/# /d;}" -e "2{/^$/d;}" -e "/# \[$(git describe --abbrev=0 --exclude="$(git describe --abbrev=0 --match='v*.*.*' --tags)" --match='v*.*.*' --tags | tr -d v)/q;p" CHANGELOG.md > /tmp/RELEASE-NOTES.md
+      - name: Release Protocol Versions
+        run: |
+          manifestFile=${{ github.event.repository.name }}_manifest.json
+          if [[ -f  $manifestFile ]] && [[ $(jq '.metadata.protocol_versions' $manifestFile | jq length) -gt 0 ]]; then
+            versions=`jq '.metadata.protocol_versions' $manifestFile | jq 'sort'`
+            protoV=`echo $versions | jq '.[0]'`
+            protoVs=`echo $versions | jq 'join(", ")'`
+
+            echo "PROTOCOL_VERSION=${protoV}" >> $GITHUB_ENV
+            echo "PROTOCOL_VERSIONS=${protoVs}" >> $GITHUB_ENV
+          fi
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.TF_PROVIDER_RELEASE_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.TF_PROVIDER_RELEASE_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+          role-to-assume: ${{ secrets.TF_PROVIDER_RELEASE_AWS_ROLE_ARN }}
+          role-duration-seconds: 3600
+          # TODO: allow session tagging once IAM permission changes addressed
+          # Reference: https://github.com/hashicorp/hc-releases/issues/124
+          role-skip-session-tagging: true
+      - name: goreleaser release
```

Comment:
so one thing we're not doing in the `awscc` provider is uploading the manifest.json to the release artifacts for each new release...which from my understanding is _ok_ since the protocol version and versions are sent through the `hc-release upload-file` header flags for each provider binary pushed to S3. nevertheless, a good example to look at  for manifest file copying is seen here https://github.com/bflad/ghaction-terraform-provider-release/blob/55a0972946e4619847280e1057cee14b947247cd/.github/workflows/community.yml#L41-L42. so before this `gorelease release` step, the manifest could be copied so that the new file has the new release version in the filename. To make sure `goreleaser` doesn't complain about this new file, you'll want to add `*_manifest.json` to the repo's .gitignore as mentioned in https://github.com/hashicorp/terraform-plugin-framework/issues/87#issuecomment-944713973